src/Application.php

```diff
@@ -1,35 +1,172 @@
+use DepDoc\PackageManager\NodePackageManager;
+use DepDoc\Parser\AbstractParser;
+use DepDoc\Parser\MarkdownParser;
+use DepDoc\Validator\PackageValidator;
+use DepDoc\Writer\AbstractWriter;
+use DepDoc\Writer\MarkdownWriter;
+
 class Application
 {
-    public function updateAction()
+    /** @var ComposerPackageManager */
+    protected $managerComposer;
+    /** @var NodePackageManager */
+    protected $managerNode;
+    /** @var AbstractParser */
+    protected $parser;
+    /** @var AbstractWriter */
+    protected $writer;
+    /** @var PackageValidator */
+    protected $validator;
+
+    public function __construct()
+    {
+        $this->managerComposer = new ComposerPackageManager();
+        $this->managerNode = new NodePackageManager();
+        $this->parser = new MarkdownParser();
+        $this->writer = new MarkdownWriter();
+    }
+
+    public function updateAction(): bool
     {
-        $composer = new PackageManager\Composer();
+        $composer = $this->getManagerComposer();
+        $node = $this->getManagerNode();
+
         $installedPackages[$composer->getName()] = $composer->getInstalledPackages();
-        $node = new PackageManager\Node();
         $installedPackages[$node->getName()] = [];
         $installedPackages[$node->getName()] = $node->getInstalledPackages();
 
-        $markdownParser = new Parser\Markdown();
-        $documentedDependencies = $markdownParser->getDocumentedDependencies();
+        $documentedDependencies = $this->getParser()
+            ->getDocumentedDependencies();
 
-        $writer = new Writer\Markdown();
-        $writer->createDocumentation($installedPackages, $documentedDependencies);
+        $this->getWriter()
+            ->createDocumentation($installedPackages, $documentedDependencies);
+
+        return true;
     }
 
-    public function validateAction()
+    public function validateAction(): bool
     {
-        $composer = new PackageManager\Composer();
+        $composer = $this->getManagerComposer();
+        $node = $this->getManagerNode();
+
         $installedPackages[$composer->getName()] = $composer->getInstalledPackages();
-        $node = new PackageManager\Node();
         $installedPackages[$node->getName()] = $node->getInstalledPackages();
 
-        $markdownParser = new Parser\Markdown();
-        $documentedDependencies = $markdownParser->getDocumentedDependencies();
+        $documentedDependencies = $this->getParser()
+            ->getDocumentedDependencies();
+
+        $validationResult = $this->getValidator()
+            ->compare($installedPackages, $documentedDependencies);
+
+        if (empty($validationResult)) {
+            return true;
+        }
+
+        foreach ($validationResult as $line) {
+            echo $line  . PHP_EOL;
+        }
+
+        return false;
+    }
+
+    /**
+     * @return ComposerPackageManager
+     */
+    public function getManagerComposer(): ComposerPackageManager
```

Comment:
Welchen Vorteil bringen die Getter?

Reply:
Die getter werden, wenn der PR durch ist, auch innerhalb der Klasse genutzt, damit man die Abhängigkeit von außen ändern kann.
Also ja, @cheeZery :D

Reply:
Für das ändern benötigt man aber keine Getter. Da reichen die Setter. Die Klasse hat keine Notwendigkeit ihre Abhängigkeiten nach außen zu promoten.

Reply:
Im Unit Test kann ich aber keine Properties mocken.
Sobald es in der `Application` Klasse eine nicht.öffentliche Methode gibt, die eine der Eigenschaften nutzt, muss das über den Getter passieren. Sonst gibt es beim Test der `Runner` Klasse (aktuell die einzige, die die `Application` nutzt, einen Fehler.
Deswegen generiere ich für gewöhnlich immer Getter und Setter für die Properties.
Da es sich nur um einen Getter handelt, sehe ich hier auch keine Sicherheitsbedenken.